internal/idp/providers/ldap/session.go

```diff
@@ -142,21 +144,32 @@ func tryLogin(
 
 	user := sr.Entries[0]
 	// Bind as the user to verify their password
-
-	if err = l.Bind(user.DN, password); err != nil {
+	if err = conn.Bind(user.DN, password); err != nil {
 		return nil, ErrFailedLogin
 	}
 	return user, nil
 }
 
 func queriesAndToSearchQuery(queries ...string) string {
+	if len(queries) == 0 {
+		return ""
+	}
+	if len(queries) == 1 {
+		return queries[0]
+	}
```

Comment:
please add some tests for these functions

Reply:
and i guess there's a better option to build the strings, please at least initialize the used array with a size